File: src/main/java/com/brasil/transparente/api/entity/Ministerio.java
package com.brasil.transparente.api.entity;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.*;

import java.util.List;

@Builder
@Getter
@Setter
@Entity
@Table(name = "ministerio")
public class Ministerio {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long ministerioId;

    private String nameMinisterio;

    @ManyToOne
    @JsonBackReference
    @JoinColumn(name = "poder_id", referencedColumnName = "poderId")
    private Poder poder;

    @OneToMany(mappedBy = "ministerio", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonManagedReference
    private List<Orgao> listOrgao;

    private double totalValueSpent;

    private double percentageOfTotal;


}

File: src/main/kotlin/com/brasil/transparente/api/entity/Ministerio.kt
package com.brasil.transparente.api.entity

import com.fasterxml.jackson.annotation.JsonBackReference
import com.fasterxml.jackson.annotation.JsonManagedReference
import jakarta.persistence.*

@Entity
@Table(name = "ministerio")
class Ministerio {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var ministerioId: Long? = null

    var nameMinisterio: String? = null

    @ManyToOne
    @JsonBackReference
    @JoinColumn(name = "poder_id", referencedColumnName = "poderId")
    var poder: Poder? = null

    @OneToMany(mappedBy = "ministerio", cascade = [CascadeType.ALL], orphanRemoval = true)
    @JsonManagedReference
    var listOrgao: MutableList<Orgao>? = null

    var totalValueSpent: Double? = null

    var percentageOfTotal: Double? = null

}